bin/faucet/src/main.rs

```diff
@@ -305,12 +314,23 @@ mod test {
 
         // Start the faucet connected to the stub
         let website_url = config.endpoint.clone();
-        tokio::spawn(async move {
-            run_faucet_command(Cli {
-                command: crate::Command::Start { config: config_path },
-            })
-            .await
-            .unwrap();
+
+        // Use std::thread to launch faucet - avoids Send requirements
```

Comment:
What was not Send?

Reply:
I had issue mostly related to the `TransactionExecutor`, and with `DataStore` in particular, because with the addition of the remote prover, the `async` feature was triggered and now the `DataStore` trait is async.

Reply:
Yea its the trait objects again. Can't recall if `Datastore` and `TransactionAuthenticator` are implicated by gRPC generated code or not. If they aren't, we could maybe fix them. Anyway, just noting for the record, thanks.
```
rustc: `(dyn miden_tx::DataStore + 'static)` cannot be sent between threads safely
the trait `std::marker::Send` is not implemented for
`(dyn miden_tx::DataStore + 'static)` required for 
`std::sync::Arc<(dyn miden_tx::DataStore + 'static)>` to implement `std::marker::Sync`
required for `&miden_tx::TransactionExecutor` to implement `std::marker::Send
```